apps/sharedtest/test/unit/crop_resize_rotate_test.js

```diff
@@ -4,13 +4,116 @@
 /* global parseJPEGMetadata */
 /* global getImageSize */
 /* global Downsample */
-/* global cropResizeRotate */
+/* global cropResizeRotate, JPEGParser */
 
+require('/shared/test/unit/mocks/mock_lazy_loader.js');
 require('/shared/js/blobview.js');
 require('/shared/js/media/jpeg_metadata_parser.js');
 require('/shared/js/media/image_size.js');
 require('/shared/js/media/downsample.js');
 require('/shared/js/media/crop_resize_rotate.js');
+require('/shared/js/media/jpeg-exif.js');
+
+function fetchBuffer(url) {
+  return new Promise(function(resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', url);
+    xhr.onload = function() {
+      if (xhr.status !== 200) {
+        reject(new Error('Failed with status: ' + xhr.status));
+      } else {
+        resolve(this.response);
+      }
+    };
+    xhr.onerror = xhr.ontimeout = function() {
+      reject(new Error('Failed' + xhr.status));
+    };
+    xhr.responseType = 'arraybuffer';
+    xhr.send();
+  });
+}
+
+function fetchBlob(url) {
+  return fetchBuffer(url).then(function(buffer) {
+    assert.ok(buffer);
+    return Promise.resolve(new Blob([buffer]));
+  }).catch(function(e) {
+    assert.ok(false, 'fetchBlob error' + e);
+  });
+}
+
+
+suite('Metadata copy test', function() {
+  suiteSetup(function() {
+
+    // We need the mock lazy loader for jpeg-exif.
+    window.LazyLoader = window.MockLazyLoader;
+  });
+
+
+  [
+    // this one is horizontal
+    { fileName: '/test-data/blue.jpg', dims: true, rotated: false,
+      targetSize: { width: 640, height: 480 } },
+    { fileName: '/test-data/blue.jpg', dims: true, rotated: false,
+      targetSize: null },
+    // this one is vertical (Exif orientation)
+    { fileName: '/test-data/red.jpg', dims: true, rotated: true,
+      targetSize: { width: 640, height: 480 } },
```

Comment:
Even though its rotated image,  please check if we need to pass output size as 480 x 640 for vertical image.